cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2463,7 +2463,7 @@ macro(build_grpc)
 endmacro()
 
 if(ARROW_WITH_GRPC)
-  set(ARROW_GRPC_REQUIRED_VERSION "1.17.0")
+  set(ARROW_GRPC_REQUIRED_VERSION "1.32.0")
```

Comment:
We should check that Homebrew and Conda, at least, have this packaged.

Reply:
Both seem fine, fortunately.

Reply:
@jduo on the mailing list, you say that it's 1.27 that is required

Reply:
Yeah that's correct. The feature is in 1.27, but in a different namespace. When I wrote this I figured for convenience of dev we'd want to use the latest stable release but now see that 1.32 isn't actually available in all our supported environments (MSYS only has 1.29). So instead what I'm planning:
- Define a macro indicating if TlsCredentialsOptions is supported. I'll see if it's possible to auto-detect this.
- Define a macro for the namespace holding TlsCredentialsOptions that is based on version.
- Compile out use of TlsCredentials based on the first macro.
- Revert changes in this PR that update the version the gRPC version 1.32.
- Change the minimum version in the codebase/CI scripts to 1.27 (only if it is already lower).